CodeGeneration/Package.swift

```diff
@@ -12,7 +12,7 @@ let package = Package(
     .executable(name: "generate-swiftsyntax", targets: ["generate-swiftsyntax"])
   ],
   dependencies: [
-    .package(url: "..", revision: "HEAD")
+    .package(path: "..")
```

Comment:
My preference would be to move this out and then have an environment variable to control it. Then the README can just have something like "Run `SOME_ENV_VAR=1 swift ..." rather than the "change the dependency from...".

Reply:
The problem with environment variables is that you can’t control them when opening the project in Xcode.

Reply:
Maybe this is a stupid idea, but how about leaving a documentation comment in the `Package.swift` file that carries the same information as the README? We could leave commented code with a dependency to the remote repository, something like this:
```swift
  dependencies: [
    .package(path: "..")
//    .package(url: "https://github.com/apple/swift-syntax", branch: "main")
  ],
```
Yes, it's not the most elegant solution, but it's practical. Developers wouldn't have to repeatedly visit the README to copy and paste this code. Considering this is just for `CodeGeneration` SwiftPM, maybe we don't need anything fancy, just something that aids in accelerating development?"

Reply:
Yes, I think leaving that information in CodeGeneration/Package.swift makes sense to me.

Reply:
Or if we would prefer to avoid commenting code we could use an enum.
```swift
fileprivate enum SwiftSyntaxDependency {
  case local
  case remote

  fileprivate var package: Package.Dependency {
    switch self {
    case .local:
      return .package(url: "..", revision: "HEAD")
    case .remote:
      return .package(url: "https://github.com/apple/swift-syntax", branch: "main")
    }
  }
}

(...)
  products: [
    .executable(name: "generate-swiftsyntax", targets: ["generate-swiftsyntax"])
  ],
  dependencies: [
    SwiftSyntaxDependency.local.package
  ],
  targets: [
    .executableTarget(
      name: "generate-swiftsyntax",
(...)
```

Reply:
> The problem with environment variables is that you can’t control them when opening the project in Xcode.

That's true, but also in that case you can still edit the package directly. So I'd still prefer it, even if we keep that in the README. I like @Matejkob suggestions, but I would still control it with an environment variable rather than hardcode to local/remote.